### ISSUE TITLE
Drop support for php 5.6 and 7.0

### ISSUE THREAD

**Issue by TheLevti** [php-lock/lock#25]
I think when php version 5.6 and 7.0 reach their end of life in the end of December of 2018 (see: http://php.net/supported-versions.php), this library should drop support for those versions in a new major version update. This will allow us to use all the new features that php 7.1+ bring. 

In my fork I already started cleaning up the project, upgrading all composer packages to their latest version and also update tests and the travis ci config.

Any thoughts on this would be appreciated.

**Comment by willemstuursma** [comment]
I agree and would welcome a PR that makes PHP 7.1 the minimum version and starts using the new features provided by this PHP version in full.

I would prefer to release this as a new minor release (1.5 / 1.6) so that it will be very easy to upgrade to the new version.

**Comment by kslr** [comment]
Has anyone started?
